vitrina/requests/views.py

```diff
@@ -44,3 +48,37 @@ def get_context_data(self, **kwargs):
         }
         context_data.update(extra_context_data)
         return context_data
+        
+        
+class RequestCreateView(LoginRequiredMixin, CreateView):
+    model = Request
+    form_class = RequestForm
+    template_name = 'base_form.html'
+
+    def form_valid(self, form):
+        self.object = form.save()
+        if self.request.user.is_authenticated:
+            self.object.user = self.request.user
+        return super().form_valid(form)
+
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+        context_data['current_title'] = _('Poreikio registravimas')
+        return context_data
+
+
+class RequestUpdateView(PermissionRequiredMixin, UpdateView):
+    model = Request
+    form_class = RequestForm
+    template_name = 'base_form.html'
+
+    def has_permission(self):
+        request = get_object_or_404(Request, pk=self.kwargs.get('pk'))
+        if self.request.user.is_authenticated:
```

Comment:
You should also add `LoginRequiredMixin` to properly handle situation when user is not authenticated. Once you have `LoginRequiredMixin`, you don't need this line, because `self.request.user` will be guaranteed to be authenticated.